build/index.js

```diff
@@ -610,25 +608,22 @@ async function buildLiveSamplePageFromURL(url) {
   if (!document) {
     throw new Error(`No document found for ${documentURL}`);
   }
-  // Convert the lower-case sampleID we extract from the incoming URL into
-  // the actual sampleID object with the properly-cased live-sample ID.
-  for (const sampleIDObject of kumascript.getLiveSampleIDs(
-    document.metadata.slug,
-    document.rawBody
-  )) {
-    if (sampleIDObject.id.toLowerCase() === sampleID) {
-      const liveSamplePage = kumascript.buildLiveSamplePage(
-        document.url,
-        document.metadata.title,
-        (await kumascript.render(document.url))[0],
-        sampleIDObject
-      );
-      if (liveSamplePage.flaw) {
-        throw new Error(liveSamplePage.flaw.toString());
-      }
-      return liveSamplePage.html;
+  const liveSamplePage = kumascript
+    .buildLiveSamplePages(
+      document.url,
+      document.metadata.title,
+      (await kumascript.render(document.url))[0],
+      document.rawBody
+    )
+    .find((page) => page.id.toLowerCase() == sampleID);
```

Comment:
I was trying to sole the issue https://github.com/mdn/yari/issues/4436. At my PR #4437 I [add](https://github.com/mdn/yari/pull/4437/files#diff-b45a82899e064eecd30590584dc256fbf4aecbf4a782d303c94091b56d659a50L633-R634
) DecodeURI:
https://github.com/mdn/yari/blob/d0c3939267ba82bfc9e11ebaa5857ddc16c6a5c3/build/index.js#L634

I suggest
`const { safeDecodeURIComponent } = require("../kumascript/src/api/util");`
```suggestion
    .find((page) => page.id.toLowerCase() == safeDecodeURIComponent(sampleID));
```

Reply:
@shapkarin Thank you for this suggestion! I added your suggestion via https://github.com/mdn/yari/pull/3973/commits/c1db1d2c0babd2c4e4e8a4a33585f8272eddc756 but I needed to lower-case the result as well.